Scripts/RemediationScripts/Remediate-SetClassicFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1020 @@
+    $FrontDoorResources = @()
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources=@()
+
+    $controlIds = "Azure_FrontDoor_DP_Use_Secure_TLS_Version_Trial"
+    
+    # No file path provided as input to the script. Fetch all Front Doors in the Subscription.
+    if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
+        Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
+        Write-Host $([Constants]::SingleDashLine)
+        return
+        }
+        Write-Host "Fetching all Front Doors failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+        
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No  Front Door(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        $validResources | ForEach-Object { 
+            try
+            {
+            $name = $_.ResourceId.Split('/')[8]
+            $resFrontDoor = Get-AzFrontDoor -Name $name -ErrorAction SilentlyContinue
+            $FrontDoorResources = $FrontDoorResources + $resFrontDoor
+            }
+            catch
+            {
+            Write-Host "Valid resource id(s) not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host "Skipping the Resource: [$($_.ResourceName)]..."
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Valid resource id(s) not found in input json file.")    
+            $logSkippedResources += $logResource
+            Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+    }
+    else
+    {
+    # Fetch all front doors in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            Write-Host "Fetching all Front Doors in Subscription: [$($context.Subscription.SubscriptionId)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            # Get all Front Doors in a Subscription
+            $FrontDoorResources = Get-AzFrontDoor -ErrorAction Stop
```

Comment:
There is no parameter for subscriptionid

Reply:
This command does not have parameter for subscription Id.